File: backend/Api/Data/Configurations/Customer.cs
using Microsoft.EntityFrameworkCore;
using Microsoft.EntityFrameworkCore.Metadata.Builders;
using Api.Models;
using Api.Enums;

namespace Api.Data.Configurations
{
    public class CustomerConfiguration : IEntityTypeConfiguration<Customer>
    {
        public void Configure(EntityTypeBuilder<Customer> builder)
        {
            builder.ToTable("Customers");
            builder.HasKey(c => c.Id);
            builder.Property(c => c.Name).IsRequired().HasMaxLength(200);
            builder.Property(c => c.Phone).HasMaxLength(20);
            builder.Property(c => c.Email).HasMaxLength(100);
            builder.Property(c => c.Address).HasMaxLength(500);
            builder.Property(c => c.CustomerType).IsRequired();
            builder.Property(c => c.IsActive).IsRequired().HasDefaultValue(true);
            builder.Property(c => c.CreatedAt).IsRequired();

            builder.HasMany(c => c.Sales)
                .WithOne(s => s.Customer)
                .HasForeignKey(s => s.CustomerId)
                .OnDelete(DeleteBehavior.Restrict);

            builder.HasMany(c => c.Deliveries)
                .WithOne(d => d.Customer)
                .HasForeignKey(d => d.CustomerId)
                .OnDelete(DeleteBehavior.Restrict);
        }
    }
}

File: frontend/src/app/core/domain/customer/customer.ts
import { CustomerType } from '../../enums/customer-type';

/**
 * Represents a Customer
 */
export interface Customer {
  /** ID */
  id: number;

  /** Name - customer name */
  name: string;

  /** Email - customer email */
  email?: string | null;

  /** Phone - customer phone */
  phone?: string | null;

  /** Address - customer address */
  address?: string | null;

  /** Customer Type */
  customerType: CustomerType;

  /** Is Active - is the customer active */
  isActive?: boolean;

  /** Created At */
  createdAt?: Date;

  /** Updated At */
  updatedAt?: Date;

  /** Deleted At */
  deletedAt?: Date | null;
}
